lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -147,212 +134,236 @@ class ConvertAtenViewOp : public OpConversionPattern<AtenViewOp> {
       return rewriter.notifyMatchFailure(
           op, "desired size list length mismatches with the result type rank");
     }
-    SmallVector<Value> inputSizeTorchInt = getTensorSizes(rewriter, loc, input);
-    ArrayRef<Value> expandedShapeTorchInt =
-        llvm::makeArrayRef(isCollapse ? inputSizeTorchInt : outputSizeInt);
-    ArrayRef<Value> collapsedShapeTorchInt =
-        llvm::makeArrayRef(isCollapse ? outputSizeInt : inputSizeTorchInt);
-
-    // Iterate through the view op size list to do the following:
-    //
-    // 1. Combine output size list and input tensor type info to get the most
-    // static outputShape.
-    //
-    // 2. Fill in the reassociation for size list item where the output dim size
-    // is got from `torch.aten.size.int(inputTensor, inputDim)`. We naively
-    // assume this means the corresponding dimension is not expanded or
-    // collapsed. Note this may technically not always be true.
-    // TODO: think of a way better way to at least detect when this assumption
-    // is violated.
-    SmallVector<int64_t> outputShape(resultRank, kUnknownSize);
-    SmallVector<ReassociationIndices> reassociation(collapsedRank);
-    llvm::Optional<int64_t> inferredDimension;
-    for (auto en : llvm::enumerate(outputSizeTorchInt)) {
-      int64_t inputDim;
-      int64_t size;
-      int64_t outputDim = en.index();
-      // Match torch.aten.size.int(inputTensor, inputDim) with constant inputDim
-      if (matchPattern(en.value(),
-                       m_TorchTensorSizeInt(op.self(), &inputDim))) {
-        auto collapsedDim = isCollapse ? outputDim : inputDim;
-        auto expandedDim = isCollapse ? inputDim : outputDim;
-        reassociation[collapsedDim].push_back(expandedDim);
-        if (!inputType.isDynamicDim(inputDim)) {
-          outputShape[outputDim] = inputShape[inputDim];
-          continue;
-        }
-      } else if (matchPattern(en.value(), m_TorchConstantInt(&size))) {
-        if (size != -1) {
-          outputShape[outputDim] = size;
-          continue;
-        }
+    SmallVector<Value> inputSize = getTensorSizes(rewriter, loc, input);
 
-        if (inferredDimension.hasValue()) {
-          return rewriter.notifyMatchFailure(
-              op, "at most one element in size list is allowed to be -1");
-        }
-        inferredDimension = outputDim;
+    Value result;
+    // For the case when inputRank == resultRank, we only have support for the
+    // cases where not only the ranks of the input and result are equal but also
+    // the shapes. If the shape of the input tensor is static, then this case is
+    // automatically handled by PyTorch, but we need this code when the shape of
+    // the input is dynamic. The result here is the input with the shape
+    // transformed to static from dynamic.
```

Comment:
nit: The result is not being transformed from dynamic to static. All that is happening is that checks are made during runtime.